File: examples/test_simple_annotated.py
#!/usr/bin/env python3
"""Simple test for annotated assignments without values."""

def test_annotated_assignments():
    # Basic types without values
    counter: int
    name: str
    is_valid: bool
    score: float
    
    # Now assign values
    counter = 10
    name = "test"
    is_valid = True
    score = 3.14
    
    # Use them
    print(counter)
    print(name)
    print(is_valid)
    print(score)

if __name__ == "__main__":
    test_annotated_assignments()

File: examples/test_simple_annotated.rs
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_annotated_assignments() {
    let mut counter: i32 = 0;
    let mut name: String = "".to_string();
    let mut is_valid: bool = false;
    let mut score: f64 = 0.0;
    counter = 10;
    name = "test".to_string();
    is_valid = true;
    score = 3.14;
    println!("{}", counter);
    println!("{}", name);
    println!("{}", is_valid);
    println!("{}", score);
}
